src/helpers/slack.ts

```diff
@@ -155,7 +155,7 @@ async function handleSlackJoinEvent(
 
     // substitute user id in random welcome message with real user id
     const welcomeMessageText = welcomeMessage.blocks[0].text.text.replace(
-        "user_id",
+        /user_id/g,
```

Comment:
To ensure accuracy and avoid unintended consequences, it is recommended to replace the entire object rather than just the text when using regex. By doing so, you can eliminate the risk of accidentally replacing other occurrences of user_id with userId. Based on a quick review of the code data, it appears that your intention is to replace <@user_id> specifically with the value of userId and nothing more.

`.replace(/<@user_id>/g, userId);`



eg:
```
const welcomeMessageBlocksReplaced = JSON.stringify(welcomeMessageBlocks)
  .replace(/user_id/g, userId);
```

Reply:
I have updated this logic to use `split` and `join` to replace the `userId` as discussed 👍